### PR TITLE
`Learning paths`: Hide unreleased learning objects in learning path view

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I documented the Java code using JavaDoc style.

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Learning paths should not show unreleased learning objects to the student, which is currently the case.

### Description

Learning paths no longer send data like the title to the client. Instead the unreleased learning objects are marked with a placeholder in the learning path view, so students are aware that they will get more learning opportunities later on.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Course with learning paths enabled and one Competency

1. Log in to Artemis
2. Navigate to Course Administration
3. Create an exercise/lecture unit and set the release date in the future
4. Check that it is not shown in the learning path view to the student

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| learning-path-nav-overview-learning-objects.component.ts | 85.71% | ✅ |

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| LearningPathService.java | 97% | ✅ |

### Screenshots

image

## Summary by CodeRabbit

- **New Features**
  - Introduced visibility status for learning objects, allowing users to see which materials are accessible or unreleased.
  - Added visual indicators for unreleased learning objects in the user interface, enhancing usability.

- **Bug Fixes**
  - Enhanced filtering logic to ensure only visible learning materials are presented to students.

- **Documentation**
  - Updated localization files to include labels for unreleased learning objects in both German and English.

- **Tests**
  - Added new tests to ensure unreleased learning objects do not appear in learning path navigation.

### CODE REVIEW

**Review by coolchock — DISMISSED** [review]
tested on ts4, works as expected. code also LGTM

File: src/main/java/de/tum/in/www1/artemis/service/learningpath/LearningPathService.java
Comment (MaximilianAnzinger) [inline src/main/java/de/tum/in/www1/artemis/service/learningpath/LearningPathService.java:368]: Please adjust the comment to highlight that this step also filters out unreleased units

**Review by MaximilianAnzinger — CHANGES_REQUESTED** [review]

**Review by Strohgelaender — APPROVED** [review]
code lgtm

**Review by pzdr7 — APPROVED** [review]
Works as described on TS4 👍

**Review by Strohgelaender — DISMISSED** [review]

**Review by pzdr7 — APPROVED** [review]
Reapprove after server style fix

**Review by coolchock — DISMISSED** [review]
re-approve after style changes

**Review by BBesrour — COMMENTED** [review]
code lgtm!

The server-style check is failing though

**Review by BBesrour — CHANGES_REQUESTED** [review]
view previous [comment](https://github.com/ls1intum/Artemis/pull/9105#pullrequestreview-2203217206)

**Review by Strohgelaender — APPROVED** [review]

**Review by pzdr7 — APPROVED** [review]

**Review by BBesrour — APPROVED** [review]

**Review by MaximilianAnzinger — APPROVED** [review]
Maintainer approved

**Review by Strohgelaender — APPROVED** [review]

**Review by BBesrour — APPROVED** [review]
re-approve after merge conflict

**Review by pzdr7 — APPROVED** [review]
Reapprove

**Review by coolchock — APPROVED** [review]
re-approve after merge conflicts

**Review by MaximilianAnzinger — APPROVED** [review]
Maintainer approved

**Pull request merged**
